docs/DeveloperGuide.md

```diff
@@ -80,17 +78,17 @@ is responsible for,
 
 **How the components interact with each other**
 
-The following Sequence Diagram, with the specific classes and methods abstracted, shows a high-level view on how the 
-components interact for the scenario where the user types `add /g 2 /n Alice`.
+The following Sequence Diagram shows a high-level view on how the components interact when the user enters the command
+`add /g 2 /n Alice`.
 
 ![ArchitectureSequenceDiagram](images/ArchitectureSequenceDiagram.png)
 > :information_source: **Note:** The lifeline for `AddPersonCommand` ends at the destroy marker :x:
-> but due to the limiations of PlantUML, the lifeline eraches the end of the diagram.
+> but due to the limitations of PlantUML, the lifeline reaches the end of the diagram.
 
-Each of the main components shown in the diagram above is defined and implemented in a class with the same name as its
-component. The section below provides a more in-depth details on how the components interact with one another.
+Each of the components are defined and implemented as a class with the same name. The section below provides 
+more in-depth details on how the components interact with one another.
 
-Each component may have several other classes underneath it, belonging to the same logical grouping, to reduce coupling.
+Each component may have several other classes underneath it, belonging to the same logical grouping.
```

Comment:
Consider rephrasing this portion further. What does it mean to have classes underneath the component? By means of extension, or consisting? 
`Logical grouping` here could also be paraphrased to not be confused with generation logical grouping